robot/web/web.go

```diff
@@ -260,34 +260,36 @@ func New(ctx context.Context, r robot.Robot, logger golog.Logger, opts ...Option
 }
 
 type webService struct {
-	mu           sync.Mutex
-	r            robot.Robot
-	rpcServer    rpc.Server
-	modServer    rpc.Server
-	streamServer *StreamServer
-	services     map[resource.Subtype]subtype.Service
-	opts         options
-	addr         string
-	modAddr      string
-
+	mu                      sync.Mutex
+	r                       robot.Robot
+	rpcServer               rpc.Server
+	modServer               rpc.Server
+	streamServer            *StreamServer
+	services                map[resource.Subtype]subtype.Service
+	opts                    options
+	addr                    string
+	modAddr                 string
 	logger                  golog.Logger
-	cancelFunc              func()
+	cancelFuncs             []func()
+	isRunning               bool
 	activeBackgroundWorkers sync.WaitGroup
 }
 
 // Start starts the web server, will return an error if server is already up.
 func (svc *webService) Start(ctx context.Context, o weboptions.Options) error {
 	svc.mu.Lock()
 	defer svc.mu.Unlock()
-	if svc.cancelFunc != nil {
+	if svc.isRunning {
```

Comment:
This is needed because `Update` may be called before `Start`. Previously this was not an issue since `Update` didn't set `cancelFunc`.